src/gencost/waterfall.py

```diff
@@ -1174,104 +1217,163 @@ def get_860_by_x(
             )
             .copy()
             .merge(
-                xwalk[
-                    ["plant_id_eia", "generator_id", subplant_id_col]
-                ].drop_duplicates(),
+                coi[["plant_id_eia", "generator_id", "pollution_control_costs_per_kw"]],
                 on=["plant_id_eia", "generator_id"],
-                how="outer",
+                how="left",
                 validate="m:1",
-                indicator=True,
             )
             .merge(
-                coi[["plant_id_eia", "generator_id", "pollution_control_costs_per_kw"]],
+                self.pudl_tabl.gens_eia860m()[
+                    [
+                        "plant_id_eia",
+                        "generator_id",
+                        "balancing_authority_code_eia",
+                        # "state",
+                    ]
+                ].drop_duplicates(),
                 on=["plant_id_eia", "generator_id"],
                 how="left",
                 validate="m:1",
             )
         )
 
-        test = (
-            merged.query(
-                "_merge != 'both' & prime_mover_code in @FOSSIL_PRIME_MOVER_MAP"
+        if subplant_id_col == "generator_id":
+            # have to cast True, False, and NA from tech cols in 860 as 0 and 1
+            # since at the PF level, this is done in wt avg
+
+            merged = self.tech_cols_dummy(merged)
+
+            if merge_only:
+                return merged
+
+            return (
+                merged.assign(
+                    age_from_report_year=lambda x: (
+                        x["report_date"] - x["generator_operating_date"]
+                    ).dt.days
+                    / 365.25,
+                    avg_age_from_report_year=lambda x: x.age_from_report_year,
+                    current_age=lambda x: (
+                        age_year_str - x["generator_operating_date"]
+                    ).dt.days
+                    / 365.25,
+                    current_avg_age=lambda x: x.current_age,
+                    age_of_observation=lambda x: (
+                        age_year_str - x["report_date"]
+                    ).dt.days
+                    / 365.25,
+                    age_relative_to_avg=lambda x: x["current_age"]
+                    - x["avg_age_from_report_year"],
+                )
```

Comment:
Can this part be pulled to before `if subplant_id_col == "generator_id":`? I think that would allow the same age calculations to be used for both making it easier to keep them in line.